File: No.2490-Bees-master/api/src/main/java/org/bees/biz/service/NoticeService.java
package org.bees.biz.service;

import java.lang.reflect.InvocationTargetException;
import java.util.Date;
import java.util.LinkedList;
import java.util.List;

import org.apache.commons.beanutils.PropertyUtils;
import org.apache.commons.logging.Log;
import org.apache.commons.logging.LogFactory;
import org.bees.biz.persistence.manager.NoticeManager;
import org.bees.biz.persistence.manager.UserManager;
import org.bees.biz.persistence.model.Notice;
import org.bees.biz.service.dto.NoticeDto;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class NoticeService {
	
	@Autowired
	private NoticeManager noticeManager;	
		
	@Autowired
	private UserManager userManager;
	
	

	@SuppressWarnings("unused")
	private final static Log LOG = LogFactory.getLog(NoticeService.class);

	public Notice getInfoById(Integer id) {		
		return this.noticeManager.getInfoById(id);
	}

	public List<NoticeDto> getList(String title, Byte status, Date startDate, Date endDate, Integer limit, Integer start) {
		List<Notice> notices=this.noticeManager.getList(title,status,startDate,endDate,start,limit);
		List<NoticeDto> dtos=new LinkedList<NoticeDto>();
		for (Notice notice : notices) {
			NoticeDto dto=new NoticeDto();
			try {
				PropertyUtils.copyProperties(dto, notice);
			} catch (IllegalAccessException | InvocationTargetException
					| NoSuchMethodException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			}
			dtos.add(dto);
		}		
		return dtos;
	}

	public boolean add(Integer userId, String title, String content, Byte status) {

		return this.noticeManager.add(userId,title,content,status);
	}

	public boolean del(Integer id) {
		return this.noticeManager.del(id);
	}

	public boolean edit(Integer id, String title, String content, Byte status) {
		return this.noticeManager.edit(id,title,content,status);
	}

	public Integer getListCount(String title, Byte status, Date startDate,
			Date endDate) {
		return this.noticeManager.getListCount(title, status, startDate, endDate);
	}

	
	
}

File: beesInterface/bees_notice/noticeService.py
from models import Notice
from utils.const import Const
from utils.validateUtils import ValidateUtils

class NoticeService:
	def __init__(self):
		pass
	
	@staticmethod
	def getInfoById(id):
		try:
			return Notice.objects.get(id=id, del_flag=Const.DEL_FLAG_NOT_DEL)
		except Exception, e:
			return None
		
	@staticmethod
	def getList(title, status, startDate, endDate, start, limit):
		notices = Notice.objects.filter(del_flag=Const.DEL_FLAG_NOT_DEL).order_by('-id')
		if not ValidateUtils.isNull(title):
			notices = notices.filter(title__contains=title)
		if not ValidateUtils.isNull(status):
			notices = notices.filter(status=int(status))
		if not ValidateUtils.isNull(startDate):
			notices = notices.filter(add_time__gte=startDate)
		if not ValidateUtils.isNull(endDate):
			notices = notices.filter(add_time__lte=endDate)
		if ValidateUtils.isNull(start):
			start = 0
		else:
			start = int(start)
		if ValidateUtils.isNull(limit):
			notices = notices[start:]
		else:
			limit = int(limit)
			notices = notices[start: start+limit]
		dtos = []
		for notice in notices:
			dtos.append(notice.getNoticeDto())
		return dtos
	
	@staticmethod
	def add(userId, title, content, status):
		notice = Notice()
		notice.create_id = userId
		notice.title = title
		notice.content = content
		notice.status = int(status)
		notice.save()
		return True
	
	@staticmethod
	def delete(id):
		try:
			notice = Notice.objects.get(id=id)
			notice.del_flag = Const.DEL_FLAG_DEL
			notice.save()
			return True
		except Exception, e:
			return False
	
	@staticmethod
	def edit(id, title, content, status):
		try:
			notice = Notice.objects.get(id=id)
			notice.content = content
			notice.title = title
			notice.status = status
			notice.save()
			return True
		except Exception, e:
			return False
	
	@staticmethod
	def getListCount(title, status, startDate, endDate):
		notices = Notice.objects.filter(del_flag=Const.DEL_FLAG_NOT_DEL)
		if not ValidateUtils.isNull(title):
			notices = notices.filter(title__contains=title)
		if not ValidateUtils.isNull(status):
			notices = notices.filter(status=status)
		if not ValidateUtils.isNull(startDate):
			notices = notices.filter(add_time__gte=startDate)
		if not ValidateUtils.isNull(endDate):
			notices = notices.filter(add_time__lte=endDate)
		return notices.count()
